tracer/src/Datadog.Trace/ClrProfiler/InstrumentationDefinitions.cs

```diff
@@ -11,13 +11,19 @@ internal static partial class InstrumentationDefinitions
 
         internal static Payload GetAllDefinitions()
         {
+            var definitions = GetDefinitionsArray();
+            if (ServerlessInstrumentation.Serverless.IsRunningInLambda())
+            {
+                definitions = ServerlessInstrumentation.Serverless.GetDefinitions(definitions, assemblyFullName);
+            }
+
             return new Payload
             {
                 // Fixed Id for definitions payload (to avoid loading same integrations from multiple AppDomains)
                 DefinitionsId = "FFAFA5168C4F4718B40CA8788875C2DA",
 
                 // Autogenerated definitions array
-                Definitions = GetDefinitionsArray(),
+                Definitions = definitions,
             };
         }
```

Comment:
I'm not sure of the modification in this class, I rather prefer to modify [`Instrumentation.Initialize`](https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs#L62) if we want to  add integrations dynamically on Initialization. Mostly to have a clear separation of both types of integrations.